Angular2-Starter-Project.md

```diff
@@ -0,0 +1,232 @@
+        <script>
+          System.import('app').catch(function(err){ console.error(err); });
+        </script>
+      </head>
+      
+      <!-- Display app in my-app element -->
+      <body>
+        <my-app>Loading Your App...</my-app>
+      </body>
+    </html>
+```
+In commented section 1, we loaded several libraries that improve the compatibility of this tutorial with different browsers/versions. It also imports system.src.js which is used below.
+
+In section 2, SystemJS is used to load our application and our various modules. In a production example we may want to use something else such as webpack. It was chosen here since we can use it with plunker.
+
+This is all that is required to get our simple application running. <a href="http://plnkr.co/edit/2i7Wjwd2JGj4NZtKvGD2" target="_blank">Here is a link to a plunker</a> that contains our working application. You can fork it into your own version and change whatever you'd like.
+
+<a target="_blank" href="http://plnkr.co/edit/2i7Wjwd2JGj4NZtKvGD2">VIEW APP</a>
+
+## Add Functionality and Another Component
+
+Now let's add a bit more functionality to our program. We will create a counter that let's you increment it by clicking a button.
+
+Our increment component will be its own module so that we can reuse it in later applications. 
+
+Let's name our file: app/increment-clicker.component.ts and set it up.
+```
+// app/increment-clicker.component.ts
+import { Component } from '@angular/core'; 
+
+@Component({
+    selector: 'increment-clicker',
+    template: `
+      <div className="counter">
+        <h1>{{curClicks}} clicks</h1>
+        <button type="button" (click)="incrementClicks()">Increment</button>
+      </div>
+    `,
+    styles: [`
+        .counter {
+          width: 100%;
+          margin: auto;
+          background: darkgreen;
+          border-radius: 5px;
+          color: white;
+          padding: 20px;
+          text-align: center;
+        }
+        .counter h1 {
+          margin: 0;
+          padding: 20px;
+          font-size: 36px;
+        }
+        .counter button {
+          background: #f1c40f;
+          border: 0;
+          box-shadow: 0px 5px 0px #927608;
+          padding: 20px;
+          width: 100%;
+          outline: none;
+          border-radius: 5px;
+          color: darkgreen;
+          font-weight: bold;
+        }
+        
+        .counter button:hover {
+          background: #a9890a;
+          cursor: pointer;
+        }
+    `]
+})
+
+export class IncrementClicker {
+  curClicks = 0;
+  
+  incrementClicks() {
+    this.curClicks++;
+  }  
+}
+```
+Our component structure is similar to our previous one, but we've added a bit of functionality here. You will notice our class is no longer empty. We've added a variable to track the clicks as well as a function to increment them when the user clicks the button.
+
+You can display variables from your class in your template using double brackets such as: {{variableName}}
```

Comment:
You can display ~~variables~~ properties from your class in your template using double ~~brackets~~ braces syntax such as: ~~{{variableName}}~~ `{{ variableName }}.`

You can display properties from your class in your template using double braces syntax such as: `{{ variableName }}`.